Dockerfile

```diff
@@ -0,0 +1,38 @@
+FROM python:3.12.0a7-buster as base
+
+# Install poetry
+RUN curl -sSL https://install.python-poetry.org | python3 -
+ENV PATH /root/.local/bin:$PATH
+
+# Make workspace
+WORKDIR /app
+COPY poetry.toml .
+COPY pyproject.toml .
+
+# Install poetry dependencies
+RUN poetry install
```

Comment:
A possible extension here might be to make this run this with the `--only-main` flag, so that it doesn't install your dev dependencies (e.g. pytest) in your base/production image, just to limit the vulnerabilities your app might be exposed to:
```suggestion
RUN poetry install --only main
```
If we do do this, we'd obviously want to add a `poetry install` command to the dev stage of the image so that the correct dependencies are present there